source/common/api/posix/os_sys_calls_impl.cc

```diff
@@ -404,5 +404,13 @@ SysCallIntResult OsSysCallsImpl::getifaddrs([[maybe_unused]] InterfaceAddressVec
 #endif
 }
 
+SysCallIntResult OsSysCallsImpl::getaddrinfo(const char* node, const char* service,
+                                             const addrinfo* hints, addrinfo** res) {
+  const auto rc = ::getaddrinfo(node, service, hints, res);
```

Comment:
Your call, but I'm not sure offhand what the return value of getaddrinfo is (int?) so I'd recommend being explicit about the return type instead of using auto here.